packages/react-signer/src/signers/MetaMaskSnapSigner.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017-2023 @polkadot/react-signer authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Signer, SignerResult } from '@polkadot/api/types';
+import type { SignerPayloadJSON } from '@polkadot/types/types';
+
+import { signSignerPayload } from 'azero-wallet-adapter';
+
+import { connectSnap } from '../snap';
+
+let id = 0;
+
+export class MetaMaskSnapSigner implements Signer {
```

Comment:
I found two places requiring `signRaw` where `isSnap` should be included in condition (I think it works without it too, but for consistency):

packages/page-files/src/UploadModal.tsx:131
packages/page-signing/src/Sign.tsx:53